src/main/view/screen/SettingsPanel.java

```diff
@@ -1,32 +1,392 @@
+			            		if (controlButtons.get(i) == controlButton) {
+			            			controlButtons.get(i).getStyleClass().add("buttonRed");
+			            			controlButtons.get(i).setText("Press Any Key");
+			            			waitingForPress = true;
+			            			controlWaiting = i;
+			            		}
+			            	}
+			            }
+		            }
+		        });
+				handleKeyPressWaiting();
+				controlButtons.add(controlButton);
+			}
+		}
+
+		private void handleKeyPressWaiting() {
+			root.setOnKeyReleased(new EventHandler<KeyEvent>() {
+			    @Override
+			    public void handle(KeyEvent event) {
+			    	if (waitingForPress) {
+			        	KeyCode key = event.getCode();
+			            String pressed = key.toString();
+			            controlButtons.get(controlWaiting).getStyleClass().remove("buttonRed");
+						if (currentMode == SettingsEnum.HUMAN) {
+							humanControls[controlWaiting][1] = pressed;
+						} else {
+							pandaControls[controlWaiting][1] = pressed;
+						}
+						controlButtons.get(controlWaiting).setText(pressed);
+						waitingForPress = false;
+			    	} 
+			    }
+			});
+		} 
+
+		//This is the drop down list for modifying the mode
+		private void setUpModeModifier() {
+			ObservableList<String> options = 
+				    FXCollections.observableArrayList(
+				        "CONTROL",
+				        "SHIFT",
+				        "ALT"
+				    );
+			modeModifier = new ComboBox<String>(options);
+			modeModifier.setVisibleRowCount(3);
+			modeModifier.setButtonCell(new ListCell<String>(){
+		        @Override
+		        protected void updateItem(String string, boolean empty) {
+		            super.updateItem(string, empty); 
+		            if(!(empty || string==null)){
+		                setStyle("-fx-text-fill: white");
+		                setText(string);
+		            }
+		        }
+
+		    });
+			modeModifier.setTranslateX(350);
+			modeModifier.setTranslateY(87 + SPACING);
+			settings.getChildren().add(modeModifier);
+		}
+		
+		@Override
+		public void draw(GraphicsContext g, Point screenDimensions) {
+			g.drawImage(getAssets().getImage("GAME_BACKGROUND"), 0, 0, screenDimensions.x, screenDimensions.y);
+			this.screenDimensions.x = screenDimensions.x;
+			this.screenDimensions.y = screenDimensions.y;
+			drawTopBar(g);
+			switch (currentMode) {
+				case GENERAL:
+					displayGeneralSettings(g);
+					break;
+				default:
+					displayControlSettings(g);
+			}
+		}
+
+		private void displayGeneralSettings(GraphicsContext g) {
+			loadButton.setVisible(false);
+			saveButton.setVisible(false);
+			setCommandButtonVisibility(false);
+			g.setFont(getAssets().getFont(3));
+		    g.setFill(Color.WHITE);
+		    g.setEffect(ds);
+		    g.fillText("General Settings", 15, 100);
+		    g.setEffect(null);
+		}
+
+		private void displayControlSettings(GraphicsContext g) {
+			loadButton.setVisible(true);
+			saveButton.setVisible(true);
+			setCommandButtonVisibility(true);
+			g.setFont(getAssets().getFont(3));
```

Comment:
Assets is an instance variable of the superType Panel. I don't understand how this is a TDA violation either! I'm getting a font from "Assets" and applying it to the Graphics Context "g". I'm not taking something from Assets and setting it back to Assets!